source/visionEnhancementProviders/NVDAHighlighter.py

```diff
@@ -316,50 +316,57 @@ def onPanelActivated(self):
-			settings.highlightFocus,
-			settings.highlightNavigator,
+			settingsStorage.highlightBrowseMode,
+			settingsStorage.highlightFocus,
+			settingsStorage.highlightNavigator,
 		]
-		if any(settingsToTriggerActivation):
-			if all(settingsToTriggerActivation):
-				self._enabledCheckbox.Set3StateValue(wx.CHK_CHECKED)
-			else:
-				self._enabledCheckbox.Set3StateValue(wx.CHK_UNDETERMINED)
-			self._ensureEnableState(True)
+		isAnyEnabled = any(settingsToTriggerActivation)
+		if all(settingsToTriggerActivation):
+			self._enabledCheckbox.Set3StateValue(wx.CHK_CHECKED)
+		elif isAnyEnabled:
+			self._enabledCheckbox.Set3StateValue(wx.CHK_UNDETERMINED)
 		else:
 			self._enabledCheckbox.Set3StateValue(wx.CHK_UNCHECKED)
-			self._ensureEnableState(False)
 
-	def _ensureEnableState(self, shouldBeEnabled: bool):
+		if not self._ensureEnableState(isAnyEnabled) and isAnyEnabled:
+			self._onEnableFailure()
+
+	def _onEnableFailure(self):
+		""" Initialization of Highlighter failed. Reset settings / GUI
+		"""
+		settingsStorage = self._getSettingsStorage()
+		settingsStorage.highlightBrowseMode = False
+		settingsStorage.highlightFocus = False
+		settingsStorage.highlightNavigator = False
+		self.updateDriverSettings()
+		self._updateEnabledState()
+
+	def _ensureEnableState(self, shouldBeEnabled: bool) -> bool:
 		currentlyEnabled = bool(self._providerControl.getProviderInstance())
 		if shouldBeEnabled and not currentlyEnabled:
-			self._providerControl.startProvider()
+			return self._providerControl.startProvider()
 		elif not shouldBeEnabled and currentlyEnabled:
-			self._providerControl.terminateProvider()
+			return self._providerControl.terminateProvider()
+		return True
 
 	def _onCheckEvent(self, evt: wx.CommandEvent):
 		settingsStorage = self._getSettingsStorage()
 		if evt.GetEventObject() is self._enabledCheckbox:
-			settingsStorage.highlightBrowseMode = evt.IsChecked()
-			settingsStorage.highlightFocus = evt.IsChecked()
-			settingsStorage.highlightNavigator = evt.IsChecked()
-			self._ensureEnableState(evt.IsChecked())
+			isEnableAllChecked = evt.IsChecked()
+			settingsStorage.highlightBrowseMode = isEnableAllChecked
+			settingsStorage.highlightFocus = isEnableAllChecked
+			settingsStorage.highlightNavigator = isEnableAllChecked
+			if not self._ensureEnableState(isEnableAllChecked) and isEnableAllChecked:
+				self._onEnableFailure()
 			self.updateDriverSettings()
 		else:
 			self._updateEnabledState()
+
 		providerInst: Optional[NVDAHighlighter] = self._providerControl.getProviderInstance()
 		if providerInst:
 			providerInst.refresh()
-		elif evt.IsChecked():
-			# One or more check boxes are enabled, so the provider instance must be there.
-			# Yet, there is no instance. This must be a case where initialization failed.
-			settingsStorage.highlightBrowseMode = False
-			settingsStorage.highlightFocus = False
-			settingsStorage.highlightNavigator = False
-			self.updateDriverSettings()
-			self._updateEnabledState()

```

Comment:
This code needed to be called multiple times (lines: 334, 362), so it was extracted to Line 336: `def _onEnableFailure(self):`